SDWebImage/SDWebImageWebPCoder.m

```diff
@@ -87,7 +88,15 @@ - (UIImage *)decodedImageWithData:(NSData *)data {
         WebPDemuxDelete(demuxer);
         return nil;
     }
-    CGColorSpaceRef colorSpace = [self sd_colorSpaceWithDemuxer:demuxer];
+    
+    CGColorSpaceRef colorSpace = NULL;
+    // ICC profile
```

Comment:
I disagree, add flags check is the more correct (Always do flags check before get bytes) and optimized way, always call `sd_colorSpaceWithDemuxer:demuxer` would iterate all WebP chunks and do compare, that's waste time.

Reply:
I agree with @zhongwuzw opinion.

Reply:
So..This obey that method' usage. What about change this into the code below ?


```objective-c
- (nonnull CGColorSpaceRef)sd_colorSpaceWithDemuxer:(nonnull WebPDemuxer *)demuxer CF_RETURNS_RETAINED {
uint32_t flags = WebPDemuxGetI(demuxer, WEBP_FF_FORMAT_FLAGS);
CGColorSpace colorSpace = NULL;

if (flags & ICCP_FLAG) {
   // Original logic, paste here
} else {
    colorSpace = SDCGColorSpaceGetDeviceRGB();
    CGColorSpaceRetain(colorSpace);
}
  return colorSpace;
}
```

Reply:
Full code, you can pick it here.

```
- (nonnull CGColorSpaceRef)sd_colorSpaceWithDemuxer:(nonnull WebPDemuxer *)demuxer CF_RETURNS_RETAINED {
    // WebP contains ICC Profile should use the desired colorspace, instead of default device colorspace
    // See: https://developers.google.com/speed/webp/docs/riff_container#color_profile
    uint32_t flags = WebPDemuxGetI(demuxer, WEBP_FF_FORMAT_FLAGS);
    CGColorSpace colorSpace = NULL;
    
    if (flags & ICCP_FLAG) {
        WebPChunkIterator chunk_iter;
        CGColorSpaceRef colorSpaceRef = NULL;
        
        int result = WebPDemuxGetChunk(demuxer, "ICCP", 1, &chunk_iter);
        if (result) {
            NSData *profileData = [NSData dataWithBytes:chunk_iter.chunk.bytes length:chunk_iter.chunk.size];
            colorSpaceRef = CGColorSpaceCreateWithICCProfile((__bridge CFDataRef)profileData);
            WebPDemuxReleaseChunkIterator(&chunk_iter);
        }
    }
    
    if (!colorSpaceRef) {
        colorSpaceRef = SDCGColorSpaceGetDeviceRGB();
        CGColorSpaceRetain(colorSpaceRef);
    }
    
    return colorSpaceRef;
}
```

Reply:
@dreampiggy Hey, this is just an internal helper method, can we not argue on this? we can skip call this method in advance directly if flags not hit ICCP_FLAG.

Reply:
Is not argue. Though this can be fixed once, but please, you should understand I'm maintain both 4.x && 5.0 codebase. So I'm happy if the same code or fix can applied for both branch. If there are no any effort.

In the 5.x codebase, the `sd_colorSpaceWithDemuxer` is called from 4 different part, if I seperate the check of that `if (flags & ICCP_FLAG) {...}`. I need to dunplicate the code 4 times. So I think put them together is better to help for me to maintain.

Reply:
... 5.x `SDWebImage` already moved out `WebP`, no merge operation we can do , you can do anything in `SDWebImageWebPCoder` project. Anyway, to save time, I'll update as you like.

Reply:
I'll apologize if my reply is offensive. I just want "if these changes is not so hard , please have a update".

And I think since all the official coders are in our orgnization, each of SDWebImage orgnization memeber can join and contribute. Even though I learned a lot knowledge about image decoding / encoding, everyone can make mistake or have something unknown part.

Anyway, I'll merge this now. As well as the 5.x WebP.